pkg/gatewayserver/io/io_test.go

```diff
@@ -371,4 +372,20 @@ func Test(t *testing.T) {
 			a.So(time.Since(last), should.BeLessThan, timeout)
 		})
 	}
+
+	for i, tc := range []struct {
```

Comment:
We don't need a test table for one entry. Also if the entry has one field, there's no need for a struct.